### PR TITLE
✨add safe tx relations

### PR DESCRIPTION
**Pull request by MartinGbz**
This PR adds:
- every tx fees linked to interacting with a safe for following addresses:
  - 0x329c54289Ff5D6B7b7daE13592C6B1EDA1543eD4
  - 0x57ab7ee15cE5ECacB1aB84EE42D5A9d0d8112922
- all gas from this address:
  - 0x3Cbded22F878aFC8d39dCD744d3Fe62086B76193
  => I exclude then the above address from all other gas computation to avoid double counting

Some refactoring has also been added

### CODE REVIEW

**Review by NandyBa — COMMENTED** [review]
Maybe can be good to delete the delegate file after this PR if we hard code address in index.js

File: index.js [hunk 1]
```diff
@@ -172,27 +165,173 @@ async function getSwapTovariableStats() {
...
+
+    return address;
+  };
+
+  const SAFE_PROXY_INTERFACE = [
+    // bytes4(keccak256("masterCopy()")) padded to 32 bytes
+    "0xa619486e00000000000000000000000000000000000000000000000000000000",
+  ];
+
+  const infuraProvider = new InfuraProvider(1, process.env.INFURA_API_KEY);
```
Comment (NandyBa) [inline index.js]: Please update example.env

File: index.js [hunk 2]
```diff
@@ -172,27 +165,173 @@ async function getSwapTovariableStats() {
...
+      delegates[idx].allTxsGasDeployer21 =
+        delegates[idx].allTxsGasDeployer21.add(gas);
+    });
+};
+
+const findDelegateIndex = (address, skipD21 = true) => {
+  // Find delegate index
+  let idx = -1;
+
+  // skip deployer 21 because we already taen into account all its txs
```
Comment (NandyBa) [inline index.js]: taken

**Comment by NandyBa** [comment]
Maybe also update the `to_block` to verify that all transactions are retrieved without issues.

File: index.js (hunk 1 above)
Comment (MartinGbz) [reply index.js]: Ah my bad, I wasn't using INFURA_API_KEY on my side too ahah
I'll modify it to use jsonRpcProvider with defillama RPC url instead 👍

**Comment by MartinGbz** [comment]
> Maybe also update the `to_block` to verify that all transactions are retrieved without issues.

I already test it on my side, but I keep the same to_block to show that output.json is almost the same

**Comment by NandyBa** [comment]
Check on my side and got the same output

**Comment by NandyBa** [comment]
> > Maybe also update the `to_block` to verify that all transactions are retrieved without issues.
> 
> I already test it on my side, but I keep the same to_block to show that output.json is almost the same

ok

**Review by NandyBa — APPROVED** [review]

**Pull request merged**